mne/viz/evoked.py

```diff
@@ -459,6 +432,44 @@ def _plot_lines(data, info, picks, fig, axes, spatial_colors, unit, units,
                 useblit=blit, rectprops=dict(alpha=0.5, facecolor='red'))
 
 
+def _handle_spatial_colors(locs3d, info, idx, chs, type, psd, ax):
+    """Helper for spatial colors."""
+    x, y, z = locs3d.T
+    colors = _rgb(x, y, z)
+    if type in ('meg', 'mag', 'grad', 'eeg'):
+        layout = find_layout(info, ch_type=type, exclude=[])
+    else:
+        layout = find_layout(info, None, exclude=[])
+    head_pos = None
+    outlines = 'skirt'
+    # drop channels that are not in the data
+    used_nm = np.array(_clean_names(info['ch_names']))[idx]
+    if layout is None:
+        name_idx = list()
+    else:
+        names = np.asarray([name for name in used_nm if name in layout.names])
+        name_idx = [layout.names.index(name) for name in names]
+    if len(name_idx) < len(chs):
+        warn('Could not find layout for all the channels. Generating custom '
+             'layout from channel positions.')
```

Comment:
Yes, that would create the layout, but we also need to set the params for ``_check_layout``. If we call the ``find_layout`` with ``on_failure='generate'``, it would pass silently.